src/modules/peek/Peek.FilePreviewer/Previewers/SvgPreviewer/SvgPreviewer.cs

```diff
@@ -0,0 +1,165 @@
+using Peek.Common.Extensions;
+using Peek.Common.Helpers;
+using Peek.Common.Models;
+using Windows.Foundation;
+
+namespace Peek.FilePreviewer.Previewers
+{
+    public partial class SvgPreviewer : ObservableObject, ISvgPreviewer, IDisposable
+    {
+        [ObservableProperty]
+        private ImageSource? preview;
+
+        [ObservableProperty]
+        private PreviewState state;
+
+        [ObservableProperty]
+        private Size imageSize;
+
+        [ObservableProperty]
+        private double scalingFactor;
+
+        public SvgPreviewer(IFileSystemItem file)
+        {
+            Item = file;
+            Dispatcher = DispatcherQueue.GetForCurrentThread();
+        }
+
+        private IFileSystemItem Item { get; }
+
+        private DispatcherQueue Dispatcher { get; }
+
+        private Task<bool>? LowQualityThumbnailTask { get; set; }
+
+        private Task<bool>? FullQualityImageTask { get; set; }
+
+        private bool IsFullImageLoaded => FullQualityImageTask?.Status == TaskStatus.RanToCompletion;
+
+        public static bool IsFileTypeSupported(string fileExt)
+        {
+            return fileExt == ".svg";
+        }
+
+        public void Dispose()
+        {
+            GC.SuppressFinalize(this);
+        }
+
+        public async Task<Size?> GetPreviewSizeAsync(CancellationToken cancellationToken)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+            var size = await Task.Run(Item.GetSvgSize);
+            if (size != null)
+            {
+                ImageSize = size.Value;
+            }
+
+            return ImageSize;
+        }
+
+        public async Task LoadPreviewAsync(CancellationToken cancellationToken)
+        {
+            State = PreviewState.Loading;
+
+            LowQualityThumbnailTask = LoadLowQualityThumbnailAsync(cancellationToken);
+            FullQualityImageTask = LoadFullQualityImageAsync(cancellationToken);
+            cancellationToken.ThrowIfCancellationRequested();
+
+            await Task.WhenAll(LowQualityThumbnailTask, FullQualityImageTask);
+
+            if (Preview == null && HasFailedLoadingPreview())
+            {
+                State = PreviewState.Error;
+            }
+        }
+
+        public async Task CopyAsync()
+        {
+            await Dispatcher.RunOnUiThread(async () =>
+            {
+                var storageItem = await Item.GetStorageItemAsync();
+                ClipboardHelper.SaveToClipboard(storageItem);
+            });
+        }
+
+        partial void OnPreviewChanged(ImageSource? value)
+        {
+            if (Preview != null)
+            {
+                State = PreviewState.Loaded;
+            }
+        }
+
+        private Task<bool> LoadLowQualityThumbnailAsync(CancellationToken cancellationToken)
+        {
+            return TaskExtension.RunSafe(async () =>
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+
+                // TODO: get thumbnail
```

Comment:
nit: can you link to a task from the board?

Reply:
I've created one: https://github.com/microsoft/PowerToys/issues/24405